Online Evals Help Doc (#433)
vellum-ai/vellum-client-generator: fern/docs/content/help/evaluation/online-evaluations.mdx

  For information on configuring metrics, see the [Out-of-the-Box Metrics](/help/metrics/out-of-the-box-metrics) section.
</Note>

## Step 3: Understanding Online Evaluations

Online Evaluations offer several key benefits for LLM application development:

1. **Real-time Performance Monitoring**: Continuously assess your deployment's performance as it handles live requests.
2. **Quality Assurance**: Ensure your LLM application maintains high standards even as input patterns may shift over time.
3. **Regression Detection**: Quickly identify any degradation in performance, allowing for swift corrective action.
4. **Insight-Driven Improvement**: Use the gathered data to inform future iterations and improvements of your LLM application.

# Viewing Evaluation Results

To access your Online Evaluation results:

1. Go to your deployment's details page.
2. Navigate to the "Executions" tab.
3. Click on an individual execution ID to view its details.
4. In the execution details, you'll find the evaluation results based on your configured metrics.

> You can analyze these results to gain insights into your LLM's strengths and areas for improvement.

Reviewer: I'd change LLM to deployment here

---
title: Leveraging Online Evaluations for LLM Development with Vellum
description: Learn how Vellum's Online Evaluations feature continuously assesses LLM outputs, ensuring model quality across diverse deployment scenarios.
---

Online Evaluations in Vellum provide a powerful way to continuously assess the quality of your deployed LLM applications. 
This feature allows you to monitor and evaluate the performance of your prompts or workflows in real-time as they're being used in production.

# Getting Started with Online Evaluations

## Step 1: Create and Deploy Your LLM Application

> 1. Start by creating either a workflow in the Workflow Sandbox or a prompt in the Prompt Sandbox.

Reviewer 2: [nit] we should always capitalize Vellum concepts - `Workflow` and `Prompt`

Reviewer 2: Would apply this feedback throughout

> 2. Once you're satisfied with your creation, deploy your workflow or prompt.

Reviewer 2: ```suggestion
2. Once you're satisfied with you created, deploy your workflow or prompt.
```

Author: LOOOL with what** you created, ill make the change

## Step 2: Configure Metrics

After deployment, you can configure metrics to evaluate your LLM application's performance:

1. Navigate to your deployed prompt or workflow.

> 2. Locate the "Metrics" tab, which is conveniently positioned between the "Releases" and "Monitoring" tabs.

Reviewer 2: ```suggestion
2. Locate the "Metrics" tab in the tab bar.
```

We don't know what other tabs or UI changes we'll make here and forget to update this doc

3. In the Metrics tab, configure your metrics according to your specific needs.

Once your metrics are configured, you can view how your executions are evaluated:

1. Go to the "Executions" tab of your deployment.
2. Click on an individual execution ID to view its details.

> 3. In the execution details, you'll find the evaluation results based on your configured metrics.

Reviewer: This is redundant with the below `Viewing Evaluation Results` section no?

> <Note>For detailed information on configuring metrics, refer to the Evaluations and Metrics sections in our documentation.</Note>

Reviewer: Does it make sense to make `Evaluations` and `Metrics` clickable links?

> ## Step 3: Save and Activate

Reviewer: We can probs merge this section with the above one since it's basically just "click the save button" haha

Once you've configured your metrics, save your changes. 
From this point forward, every execution of your deployment will be automatically evaluated against these metrics, providing you with ongoing, real-time performance insights.

# Understanding Online Evaluations

> Online Evaluations offer several key benefits for LLM development:

Reviewer: ```suggestion
Online Evaluations offer several key benefits for LLM application development:
```

> 1. **Real-time Performance Monitoring**: Continuously assess your model's performance as it handles live requests.

Reviewer: With workflows it could be more than just one model
```suggestion
1. **Real-time Performance Monitoring**: Continuously assess your deployment's performance as it handles live requests.
```

> 2. **Quality Assurance**: Ensure your LLM maintains high standards even as input patterns may shift over time.

Reviewer: ```suggestion
2. **Quality Assurance**: Ensure your LLM application maintains high standards even as input patterns may shift over time.
```

3. **Regression Detection**: Quickly identify any degradation in performance, allowing for swift corrective action.
4. **Insight-Driven Improvement**: Use the gathered data to inform future iterations and improvements of your LLM application.

# Viewing Evaluation Results

To access your Online Evaluation results:

1. Go to your deployment's details page.
2. Navigate to the "Executions" tab.
3. Click on an individual execution ID to view its details.
4. In the execution details, you'll find the evaluation results based on your configured metrics.

You can analyze these results to gain insights into your LLM's strengths and areas for improvement.

# Advanced Usage

## Multiple Metrics

> You can configure multiple metrics for a single deployment to evaluate different aspects of your LLM's performance. This allows for a more comprehensive assessment of your model's capabilities.

Reviewer: ```suggestion
You can configure multiple metrics for a single deployment to evaluate different aspects of your deployment's performance. This allows for a more comprehensive assessment of your deployment's capabilities.
```
